src/imap/scan_folders.rs

```diff
@@ -43,6 +43,16 @@ impl Imap {
                 }
             };
             let foldername = folder.name();
+            if get_watched_folders(context)
+                .await
+                .contains(&foldername.to_string())
```

Comment:
I was a little unsure about this: this currently does does case-sensitive comparison, just as `is_*box()` already do it. On the one hand, imap folder names should be case-insensitive, otoh we had no problems so far with case-sensitivity.

Reply:
Generally speaking, folder names are case-sensitive. We should save the names returned by the server and compare against them. So, if the server says it has `SeNt` folder and we decided to watch it, we should compare against it case-sensitively and still scan `Sent` folder if it exists separately. The only exception is INBOX, [RFC](https://tools.ietf.org/html/rfc3501) guarantees that there can't be INBOX and Inbox at the same time:

> In particular, this specification takes no position on case sensitivity in non-INBOX mailbox names.  Some server implementations are fully case-sensitive; others preserve case of a newly-created name but otherwise are case-insensitive; and yet others coerce names to a particular case. Client implementations MUST interact with any of these.

So comparing case-sensitively is the right thing to do here.

Reply:
rfc is pretty vague on case wrt mailbox names, https://tools.ietf.org/html/rfc3501#section-5.1 , however, as we got the names from the server, it is reasonable to just use them as is as long as we do not encounter issues (and we did not yet, as you stated).

another thing: i would consider moving `get_watched_folders()` outside the `for()` loop - there are 3 database reads in the function, for ppl using lots of folders, that may sum up.

Reply:
+1 on moving database reads outside the loop. Reads are way less expensive than writes, but this also reduces the change of returning an error in the middle of the loop, scanning some folders and failing to scan the rest.

Reply:
done